browser-sign-in/OktaBrowserSignIn/AppDelegate.swift

```diff
@@ -15,28 +15,78 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
-
+    var configForUITests: [String: String]? {
+        let env = ProcessInfo.processInfo.environment
+        guard let oktaURL = env["OKTA_URL"], oktaURL.count > 0,
+              let clientID = env["CLIENT_ID"],
+              let redirectURI = env["REDIRECT_URI"],
+              let logoutRedirectURI = env["LOGOUT_REDIRECT_URI"] else {
+            return nil
+        }
+        return ["issuer": "\(oktaURL)/oauth2/default",
+                "clientId": clientID,
+                "redirectUri": redirectURI,
+                "logoutRedirectUri": logoutRedirectURI,
+                "scopes": "openid profile offline_access"
+        ]
+    }
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        
+        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
+            try? Keychain.Search().delete()
+        }
+        window?.rootViewController = self.getRootViewController()
+        window?.makeKeyAndVisible()
         return true
     }
     
-    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
-        return false
+    func getRootViewController() -> UIViewController? {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let _ = Credential.default else {
```

Comment:
This would be clearer to guard `Credential.default != nil`, instead of assigning to `_`.

Also it might be nice to escape early with a simpler code-path.  Since the condition where `default != nil` has much more code, it might be simpler to invert this expression.

```swift
guard Credential.default == nil else {
    return storyboard.instantiateViewController(withIdentifier: "Profile")
}

// The rest of the code
```